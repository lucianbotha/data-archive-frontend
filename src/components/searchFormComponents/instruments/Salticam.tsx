import * as React from "react";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";
import { ISalticam } from "../../../utils/ObservationQueryParameters";

interface ISalticamProps {
  salticam: ISalticam;
  onChange: (value: any) => void;
}

/**
 * A form for selecting Salticam-related search parameters.
 */
const Salticam = (props: ISalticamProps) => {
  const { onChange, salticam } = props;

  const change = (e: React.FormEvent<HTMLSelectElement>) => {
    const name = e.currentTarget.name;
    const value = e.currentTarget.value;
    onChange({
      ...salticam,
      [name]: value
    });
  };

  const detectorModes = ["Normal", "Slot Mode", "Drift Scan", "Frame Transfer"];

  return (
    <MainGrid>
      <SubGrid>
        <p>Detector Mode</p>
        <SelectField name={"detectorMode"} onChange={change}>
          <AnyOption />
          {detectorModes.map(detectorMode => (
            <option key={detectorMode} value={detectorMode}>
              {detectorMode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};

export default Salticam;
