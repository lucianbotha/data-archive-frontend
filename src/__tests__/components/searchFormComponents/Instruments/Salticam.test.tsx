import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Salticam from "../../../../components/searchFormComponents/instruments/Salticam";

const onChange = jest.fn();
describe("Grids ", () => {
  it("should render Salticam", () => {
    expect(
      mount(
        <Salticam
          salticam={{ errors: {}, name: "Salticam" }}
          onChange={onChange}
        />
      )
    ).toBeDefined();
  });

  it("should render all grids correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(
        mount(
          <Salticam
            salticam={{ errors: {}, name: "Salticam" }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
