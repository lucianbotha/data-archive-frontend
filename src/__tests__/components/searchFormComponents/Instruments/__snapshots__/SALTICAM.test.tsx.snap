// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salticam  should render correctly 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

.c1.c1 {
  text-align: left;
  padding-left: 10px;
}

<Salticam
  onChange={[MockFunction]}
  salticam={
    Object {
      "errors": Object {},
      "name": "Salticam",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "main-grid",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="main-grid c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Detector Mode
              </p>
              <SelectField
                name="detectorMode"
                onChange={[Function]}
              >
                <div
                  className="control select "
                >
                  <select
                    name="detectorMode"
                    onChange={[Function]}
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="Normal"
                      value="Normal"
                    >
                      Normal
                    </option>
                    <option
                      key="Slot Mode"
                      value="Slot Mode"
                    >
                      Slot Mode
                    </option>
                    <option
                      key="Drift Scan"
                      value="Drift Scan"
                    >
                      Drift Scan
                    </option>
                    <option
                      key="Frame Transfer"
                      value="Frame Transfer"
                    >
                      Frame Transfer
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Salticam>
`;
